File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(sample_one, sample_two)
    raise ArgumentError,
          "Please enter strings of equal length" unless sample_one.size == sample_two.size
    
    sample_one.size.times.count { |x| sample_one[x] != sample_two[x] }
  end

end

File: javascript/hamming/hamming.js

var Hamming = function () {}

Hamming.prototype.compute = function (strand1, strand2) {
  var strand1Array = strand1.split('')
  var strand2Array = strand2.split('')
  var count = 0
  if (strand1Array.length !== strand2Array.length) {
    throw 'DNA strands must be of equal length.'
  }

  for (var i = 0; i <= strand1Array.length; i++) {
    if (strand1Array[i] !== strand2Array[i]) {
      count = count + 1
    }
  }
  return count
}

module.exports = Hamming
